Dockerfile.build

```diff
@@ -1,13 +1,12 @@
-FROM fedora:latest
+FROM fedora:36
```

Comment:
Can we not change this, from the past I remember that we lagged behind if we used a particular version, that's why latest was meant to solve it.

Reply:
At the moment, `latest` points to the `35` build, which doesn't have Go 1.17 available. Not sure what we could do to mitigate this if we want to stick to the fedora image, we need `36` at least but that does mean we have it tagged to a specific build

Reply:
Since we rely on a specific golang version which frequently is ahead of fedora provided version it seems reasonable to manually install specific version of golang, wdyt?

Reply:
Seems like a reasonable approach, though in that case I don't think we will be able to install Go via dnf as we have with the other packages. To ensure we can install the desired version we will have to use the manual installation outlined in https://go.dev/doc/install, do you think that's reasonable?